src/contracts/fund/trading/transactions/takeOrderOnKyber.ts

```diff
@@ -0,0 +1,90 @@
+import * as web3Utils from 'web3-utils';
+import { QuantityInterface } from '@melonproject/token-math';
+import { Contracts, Exchanges } from '~/Contracts';
+import { getExchangeIndex } from '../calls/getExchangeIndex';
+import { emptyAddress } from '~/utils/constants/emptyAddress';
+import {
+  GuardFunction,
+  PrepareArgsFunction,
+  transactionFactory,
+} from '~/utils/solidity/transactionFactory';
+import { getHub } from '../../hub/calls/getHub';
+import { getRoutes } from '../../hub/calls/getRoutes';
+import { ensureSufficientBalance } from '~/contracts/dependencies/token/guards/ensureSufficientBalance';
+import { ensureFundOwner } from '~/contracts/fund/trading/guards/ensureFundOwner';
+import { ensureIsNotShutDown } from '~/contracts/fund/hub/guards/ensureIsNotShutDown';
+import { ensureKyberTakePermitted } from '../guards/ensureKyberTakePermitted';
+import { FunctionSignatures } from '../utils/FunctionSignatures';
+
+// The order needs to be signed by the manager
+export interface TakeOrderOnKyberArgs {
+  makerQuantity: QuantityInterface;
+  takerQuantity: QuantityInterface;
+  fillTakerQuantity?: QuantityInterface;
+}
+
+const guard: GuardFunction<TakeOrderOnKyberArgs> = async (
+  environment,
+  { makerQuantity, takerQuantity, fillTakerQuantity = takerQuantity },
+  contractAddress,
+) => {
+  const hubAddress = await getHub(environment, contractAddress);
+  const { vaultAddress } = await getRoutes(environment, hubAddress);
+
+  await ensureSufficientBalance(environment, takerQuantity, vaultAddress);
+  await ensureFundOwner(environment, contractAddress);
+  await ensureIsNotShutDown(environment, hubAddress);
+
+  await ensureKyberTakePermitted(
+    environment,
+    contractAddress,
+    makerQuantity,
+    takerQuantity,
+    fillTakerQuantity,
+  );
+};
+
+const prepareArgs: PrepareArgsFunction<TakeOrderOnKyberArgs> = async (
+  environment,
+  { makerQuantity, takerQuantity, fillTakerQuantity = takerQuantity },
+  contractAddress,
+) => {
+  const exchangeIndex = await getExchangeIndex(environment, contractAddress, {
+    exchange: Exchanges.KyberNetwork,
+  });
+
+  const takerAddress = takerQuantity.token.address;
```

Comment:
think this should `takerAddress` should be `takerAsset`  and `makerAddress` should be `makerAsset`